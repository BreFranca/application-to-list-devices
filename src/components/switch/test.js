'use strict'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Tests from './tests'

Enzyme.configure({ adapter: new Adapter() })

describe('Switch should:', () => {
  const wrapperED = shallow(Tests['empty data'])
  const wrapperA = shallow(Tests['activate'])
  const wrapperD = shallow(Tests['deactivate'])
  const jExpect = global.expect

  describe('match snapshot if:', () => {
    it('isDeactivate', () => { jExpect(wrapperA).toMatchSnapshot() })
    it('is disabled', () => { jExpect(wrapperD).toMatchSnapshot() })
  })

  describe('match output if:', () => {
    it('is activate status', () => {
      expect(wrapperA.find('label').prop('className')).toBe('switch switch-active')
    })

    it('is deactivate status', () => {
      expect(wrapperD.find('label').prop('className')).toBe('switch')
    })

    it('is label of activate status', () => {
      expect(wrapperA.find('label').text()).toBe('Active')
    })

    it('is label of deactivate status', () => {
      expect(wrapperD.find('label').text()).toBe('Deactivate')
    })
  })

})