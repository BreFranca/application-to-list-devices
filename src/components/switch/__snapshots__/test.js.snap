// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should: match snapshot if: is disabled 1`] = `ShallowWrapper {}`;

exports[`Switch should: match snapshot if: isDeactivate 1`] = `ShallowWrapper {}`;
